docs/book/src/developer/providers/v1.3-to-v1.4.md

```diff
@@ -24,6 +24,7 @@ maintainers of providers and consumers of our Go API.
 - `MachineHealthCheckSuccededCondition` condition type has been removed.
 - `CloneTemplate` and `CloneTemplateInput` has been removed.
 - The option `--list-images` from `init` subcommand has been removed.
+- `NewServer` has been removed.
```

Comment:
nit Can we add the packages these identifiers have been removed from here to make it easier to follow and prevent clashes? I think this is what was done to document removals in previous version e.g. https://cluster-api.sigs.k8s.io/developer/providers/v1.1-to-v1.2.html#removals

```suggestion
- `exp/runtime/server.NewServer` has been removed.
```

Reply:
@knabben I'll merge this PR already, but maybe you can add the packages to the other identifiers in this list in one of your other PRs?